### PR TITLE
ISIS: Allow removing interface

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: protocols/isis/server/server.go [hunk 1]
```diff
@@ -81,3 +83,13 @@ func netsCompatible(nets []*types.NET) bool {
 
 	return true
 }
+
+// GetInterfaces gets all configured interfaces
+func (s *Server) GetInterfaces() []string {
```
Comment (BarbarossaTM) [inline protocols/isis/server/server.go:88]: I'm wondering if this should be called `GetInterfaceNames` as that's what we return back. As all other public methods only deal with the names it's consistent though, so no strong feelings.

File: cmd/bio-rd/is-is.go [hunk 2]
```diff
@@ -34,7 +34,13 @@ func configureProtocolsISIS(isis *config.ISIS) error {
 		}
 	}
 
+	configuredInterfaces := isisSrv.GetInterfaces()
 	for _, ifa := range isis.Interfaces {
+		if strSliceContains(configuredInterfaces, ifa.Name) {
+			log.Debugf("ISIS: Interface %q is already configured", ifa.Name)
```
Comment (BarbarossaTM) [inline cmd/bio-rd/is-is.go:40]: Is this method to be intended to be idempotent?

File: protocols/isis/server/server.go (hunk 1 above)
Comment (taktv6) [reply protocols/isis/server/server.go]: Fixed.

File: cmd/bio-rd/is-is.go (hunk 2 above)
Comment (taktv6) [reply cmd/bio-rd/is-is.go:40]: That is the idea.

**Review by BarbarossaTM — APPROVED** [review]
LGTM

**Pull request merged**